core/src/vaults.ts

```diff
@@ -0,0 +1,448 @@
+    const incentiveRelativeDai = new BigNumber(incentiveRelativeDaiHex._hex)
+        .shiftedBy(-WAD_NUMBER_OF_DIGITS)
+        .multipliedBy(debtDai);
+    const incentiveConstantDai = new BigNumber(incentiveConstantDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _fetchLiquidatedParameters = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0 && vault.initialDebtDai.eq(0)) {
+        // there was a liquidation and the vault was not used again
+        const latestEvent = liquidationEvents[liquidationEvents.length - 1];
+        return {
+            liquidationDate: await fetchDateByBlockNumber(network, latestEvent.blockNumber),
+            transactionHash: latestEvent.transactionHash,
+            auctionId: `${vault.collateralType}:${new BigNumber(latestEvent.args?.id._hex).toFixed(0)}`,
+        };
+    }
+    return undefined;
+};
+export const fetchLiquidatedParameters = memoizee(_fetchLiquidatedParameters, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _enrichVaultWithTransactonInformation = async (
+    network: string,
+    vault: Vault
+): Promise<VaultTransactionNotLiquidated> => {
+    const debtDai = vault.initialDebtDai.multipliedBy(vault.stabilityFeeRate);
+    const { liquidatiorContractAddress } = await fetchCollateralLiquidationLimitsAndLiquidatorAddress(
+        network,
+        vault.collateralType
+    );
+
+    const { liquidationRatio, oracleAddress } = await fetchLiquidationRatioAndOracleAddress(
+        network,
+        vault.collateralType
+    );
+    const collateralizationRatio = vault.collateralAmount
+        .multipliedBy(vault.minUnitPrice)
+        .multipliedBy(liquidationRatio)
+        .dividedBy(debtDai);
+    const proximityToLiquidation = vault.minUnitPrice.multipliedBy(vault.collateralAmount).minus(debtDai);
+    const { transactionFeeLiquidationEth, transactionFeeLiquidationDai } = await getApproximateLiquidationFees(
+        network
+    );
+    const { nextUnitPrice, nextPriceChange, currentUnitPrice } = await getOsmPrices(
+        network,
+        oracleAddress,
+        vault.collateralType
+    );
+    let state: 'liquidatable' | 'not-liquidatable' = proximityToLiquidation.isLessThanOrEqualTo(0)
+        ? 'liquidatable'
+        : 'not-liquidatable';
```

Comment:
Instead of making it `let`, can you please split state logic into separate function(s)?

Reply:
i am not sure that it's desirable to remove `let`: since this place repeats the logic from blockchain, `state` is changed in places where the contract would throw error.

extracted to different function